<!-- This file launches the physical gripper (open/close and tactile sensors) -->
<launch>

<!-- Parameter not set correctly/fast enough? Increase delay time below... -->
<arg name="param_launch_delay_seconds" default="1"/>
<node pkg="contactile_gripper" name="set_com_port_param" type="set_launch_params.py" output="screen"/>

<!-- Launch motor -->
<node pkg="contactile_gripper" name="gripper" type="gripper_node.py" output="screen"/>

<!-- Launch tactile sensors-->
	<node pkg="papillarray_ros_v2"
	      name="papillarray_ros_v2_node"
	      type="papillarray_ros_v2_node"
		  launch-prefix="bash -c 'sleep $(arg param_launch_delay_seconds); $0 $@' ">

		<param name="hub_id" type="int" value="0"/>

		<!-- Sensor setup parameters (don't touch generally!!)-->
            <!-- COM port parameter set externally! -->
			<!-- Number of sensors being used. Value can be 1 or 2 -->
			<param name="n_sensors" type="int" value="2"/>
			<!-- Name of COM port to connect with -->
			<!-- <param name="com_port" type="string" value="/dev/ttyACM0"/> -->
			<!-- Rate of serial connection -->
			<param name="baud_rate" type="int" value="9600"/>
			<!-- Parity: 0=PARITY_NONE, 1=PARITY_ODD, 2=PARITY_EVEN -->
			<param name="parity" type="int" value="0"/>
			<!-- Number of bits in byte. Default 8 -->
			<param name="byte_size" type="int" value="8"/>
		<!-- Read behaviour -->
			<!-- Flushing flag: flush hardware input buffer when it contains too many bytes -->
			<param name="is_flush" type="bool" value="true"/>
		<!-- Sample Rate -->s
			<!-- Rate (Hz): 100, 250, 500 or 1000 -->
			<param name="sampling_rate" type="int" value="500"/>
			
    </node>


<!-- The pose node below predicts cable pose based on tactile data -->
<!-- <node pkg="contactile_gripper" name="pose" type="pose_node.py"/> -->

<node pkg="contactile_gripper" name="control" type="control_node.py" output="screen"/>

<node pkg="contactile_gripper" name="UI" type="ui_node.py" launch-prefix="gnome-terminal --" output="screen"/>

</launch>